### DISCUSSION TITLE
Running in large scale

### DISCUSSION THREAD

**Discussion by guylissak** [moj-analytical-services/splink#2173] [General]
Hi, testing Splink for only use case of deduping a dataset of 500M  that updates daily rows.
The pipelines run in databricks using heavy spark jobs. I’m just concerned about the scale.
I did a POC using spark on 1M records and it looks good as I configure blocking rules based on the skew data but wondering if the same will work on 500M.

Do I really need to train a model about all the dataset? Or can I use just 1M rows for training and predict on the entire 500M?
Also are there specific quick win advices for such scale in the configuration?
Much appreciated

**Comment by RobinL** [comment]
500 million is a pretty big linkage.  In general, it's not theoretically valid to train on a sample of 1m, but you might find it gives you reasonably decent results anyway.    You want to make sure that your sample of 1m has a decent number of matches in it, so you might not want to take a completely random sample and instead take a sample that's more likely to to include more matches 

For further advice on scaling see:
https://moj-analytical-services.github.io/splink/topic_guides/performance/drivers_of_performance.html

**Comment by guylissak** [comment]
Hi not sure i understand your first part of the answer.
1M samples are large training amount? or you mean 1M samples are not enough to train on for 500M prediction dataset?
what is the ratio you suggest between the training dataset to the validation dataset?
Also i made sure those 1m records contain duplications.

**Reply by kavinpauly12-bit** [reply]
Hi bro , I am also doing for 460M data 

first I took a sample of around 1 million and tested it

but with me the problem comes with results, the results look pretty bad 

can you help me by sharing how did you do? 

Also, if possible may I get the code of yours, I tried many runs and experimented but the results look kind of bad 

please help me with this 


Also, In databricks spark cluster, may I know what’s the cluster config you used and approx time it took to predict for your 400m data ? 

Please help me with these
